pkg/plugins/plugins.go

```diff
@@ -229,10 +229,10 @@ func (pm *PluginManager) scan(pluginDir string, requireSigned bool) error {
 		}
 
 		pm.log.Debug("Found plugin", "id", plugin.Id, "signature", plugin.Signature, "hasRoot", plugin.Root != nil)
-		if !scanner.validateSignature(plugin) {
-			pm.log.Debug("Not adding plugin since it lacks a valid signature", "id", plugin.Id,
-				"signature", plugin.Signature)
-			continue
+		signingError := scanner.validateSignature(plugin)
+		if signingError != nil {
+			pm.log.Warn("Plugin has signature errors", "id", plugin.Id,
+				"signature", plugin.Signature, "status", signingError.ErrorCode.String())
```

Comment:
Changing this will obviously result in that unsigned or invalid signature for plugins will allow frontend plugin static content to be loaded in the UI and backend plugins to start running in the backend. Returning an error here before was an easy way out to stop all of this from happening.

With current suggestion of changes I don't really see how/where we need to put code to make sure a loaded plugin cannot be used. In general I'm worried that we spread the logic of whether a plugin is able/allowed to run/be used throughout the code base. Feels like we're missing some kind of abstraction maybe?

Instead of calling `loader.Load` below if valid signature is required and there are errors can we load a "shadow plugin" instead that's just basically plugin meta data and nothing else?

Or alternatively, would it help/be easier to add a new API endpoint where you can query for failed plugins and only get back a list of basic plugin meta data such as id/name, plugin state and the reason it failed to start? If so a different map of failed plugins can be kept in the backend by the plugins package? A bit hard to know if this is possible without knowing what the idea with the UI is.

Reply:
I share @marefr's concerns. Are they still valid?